### ISSUE TITLE
docker-desktop-proxy failed to start

### ISSUE THREAD

**Issue by zanettea** [docker/for-win#8316]
- [x] I have tried with the latest version of my channel (Stable or Edge)
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: FFA3E00D-5E97-41EE-B14E-3B11DEB1E08C/20200902083611

### Actual behavior

Docker-desktop-proxy failed to start with out-of-memory while executing restoreMounts

### Information
The problem appeared updating to 2.3.5.1. Problem is visible with 2.3.6.0. Latest stable 2.3.0.4 is working properly.

Windows 10, build 20185.re_prerelease.200731-1415

It seems docker is trying to mount recursively the same folders.  Indeed I have a huge mtab file, with entries like the one below:

`/dev/sdd /tmp/docker-desktop/shadowing-mounts/ea25db68369ea27f1a849d85588fc9fac5b760e9ada8b16c71919f2ed1e58587/tmp/docker-desktop/shadowing-mounts/58142f3d8282f3aea1a2d0b4ca1ca021fdc42dc31c4b41a935361afee82b955b/tmp/docker-desktop/shadowing-mounts/5e07ef40ff498a15484c98eb7a2616d2bd7a4dbfe5e523d0186d058d7031be2d/tmp/docker-desktop/shadowing-mounts/b1a54cefe4f7a3a8c46ba8b11c3b677da2e10250197c8ae13606e4af7ebc51ee/tmp/docker-desktop/shadowing-mounts/11b71eb5ef84a814e096315c9900b62c8f8aabc5943bfc421841f0a6a6bc1890/tmp/docker-desktop/shadowing-mounts/ea355cb4a36451dea09c536f46ba4a4b1036e53bca7b5be624399588d2ab8c1e/tmp/docker-desktop/shadowing-mounts/2d201006704e6606479c44be997092e29d7c5340f1754e06e8cc94f2637f1655/tmp/docker-desktop/shadowing-mounts/cf0af6db302b778e9c64fc940dd9d2f55ad9baf65c5f42d242a366e3c4dd8d3e/tmp/docker-desktop/shadowing-mounts/72fc7827dd8592bdccd6d343d3d9dc819baab1d2d5130c218d321b1effa75a47/tmp/docker-desktop/shadowing-mounts/23ca147adfb828c9479b4c280c968518b441ecc5dec4e638834b50b0356e3e1c/tmp/docker-desktop/shadowing-mounts/3c9cb28ca61e6b2b86ee202d4296b08c8fd1eb53fdbbb28f7e704cc8c77ece97/tmp/docker-desktop/shadowing-mounts/ea25db68369ea27f1a849d85588fc9fac5b760e9ada8b16c71919f2ed1e58587 ext4 rw,relatime,discard,errors=remount-ro,data=ordered 0 0
`

**Comment by simonferquel** [comment]
Hi,
2.3.6.0 is supposed to have fixed the issue, but you might need to run `wsl --shutdown` first. 
However, what 2.3.6.0 does is a workaround for an issue within wslpath utility, which behavior might have changed in build 20185. I'll check on an insider build to see if I can reproduce.

**Comment by simonferquel** [comment]
Hi there, sorry for the last few edge releases: we tried to fix an issue with mounts and WSL2 preventing people to do things like `-v /:/somwhere` without breaking Docker Desktop. The approach we took in versions 2.3.5.x and 2.3.6.0 proved to break even more people... 
So we took tried to solve the issue in a completely different way for the next Edge release... in the meantime I have a private build, I'd really like you to try out with this new Mounts fix: https://desktop-stage.docker.com/win/edge/47743/Docker%20Desktop%20Installer.exe. Thanks for your continued feedback!
